File: 프로그래머스/lv3/42884. 단속카메라/단속카메라.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool cmp(vector<int> &lhs, vector<int> &rhs) {
    return lhs[1] < rhs[1];
}

int solution(vector<vector<int>> routes) {
    int answer = 0;
    for(vector<int> route: routes) {
        if(route[0] > route[1]) {
            int tmp = route[0];
            route[0] = route[1];
            route[1] = tmp;
        }
    }
    sort(routes.begin(), routes.end(), cmp);
    
    int curIdx = 0;
    while (curIdx < routes.size()) {
        int endPoint = routes[curIdx][1];
        curIdx++;
        while(curIdx < routes.size() && endPoint >= routes[curIdx][0]) {
            curIdx++;
        }
        answer++;
    }
    
    return answer;
}

File: 프로그래머스/3/42884. 단속카메라/단속카메라.py
def solution(routes):
    answer = 0
    
    for route in routes:
        if route[0] > route[1]:
            route[0], route[1] = route[1], route[0]
            
    routes.sort(key=lambda x: x[1])
    
    idx = 0
    length = len(routes)
    while idx < length:
        end = routes[idx][1]
        
        idx += 1
        while idx < length and routes[idx][0] <= end:
            idx += 1
        
        answer += 1

    return answer